File: 9.palindrome-number.py
#
# @lc app=leetcode id=9 lang=python
#
# [9] Palindrome Number
#

# @lc code=start
import math


class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        n = x
        r = 0

        if x < 0:
            return False

        while n > 0:
            r = r * 10 + n % 10
            n = n // 10

        if r == x:
            return True
        return False
        # @lc code=end

File: Java/9.palindrome-number.java
/*
 * @lc app=leetcode id=9 lang=java
 *
 * [9] Palindrome Number
 */

// @lc code=start
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        // String y = new String(Integer.toString(x));
        // int lpt = 0;
        // int rpt = y.length() - 1;
        // while (lpt < rpt) {
        // if (y.charAt(lpt) != y.charAt(rpt)) {
        // return false;
        // }
        // lpt++;
        // rpt--;
        // }
        // return true;
        int temp = x;
        int reverse = 0;
        while (temp > 0) {
            int last_digit = temp % 10;
            temp = temp / 10;
            reverse = reverse * 10 + last_digit;
        }
        return reverse == x;
    }
}
// @lc code=end
